### PR TITLE
feat: cc-0043 get code from Google and get userinfo from backend

### PR DESCRIPTION
**Pull request by Jzeva**
Users can use their Google accounts to log into our website.

image
image

### CODE REVIEW

File: src/components/Login/index.tsx [hunk 1]
```diff
@@ -16,16 +16,42 @@ import MainLogoSvg from "@/assets/svg/CourtCanva-main-LOGO.svg";
 import { IconContext } from "react-icons";
 import { FaEnvelope } from "react-icons/fa";
 import { FcGoogle } from "react-icons/fc";
-import React from "react";
+import { useGoogleLogin } from "@react-oauth/google";
+import axios from "axios";
+import React, { useEffect, useState } from "react";
```
[the remark this answers is not in the document]
Comment (Jzeva) [reply src/components/Login/index.tsx:21]: Exactly, thanks for reminding.

File: src/components/Login/index.tsx [hunk 2]
```diff
@@ -16,16 +16,42 @@ import MainLogoSvg from "@/assets/svg/CourtCanva-main-LOGO.svg";
...
+          props.onClose();
+        }
+      })
+      .catch((err) => console.warn(err));
+  };
+
+  // Send authorization request to Google's Oauth server
+  const handleLogin = useGoogleLogin({
+    onSuccess: handleSuccess, // The method which would execute after authorization
+    flow: "auth-code", // OAuth 2.0 flow between implicit and authorization flow
```
[the remark this answers is not in the document]
Comment (Jzeva) [reply src/components/Login/index.tsx:52]: Yes, it should be authorization code flow. I will make the comment more explicit.

**Review by jennawan — APPROVED** [review]
Nice work! The feature works perfectly. Just reminded that userInfo will need to be stored in redux in future work.

**Comment by Jzeva** [comment]
> II have successfully authenticated my login. But I didn't get my info in the local storage, and my backend server got an error like this. Screen Shot 2022-06-28 at 12 40 07 am

Did you put the correct environment virables into the .env file?

**Review by EmilyGong111 — APPROVED** [review]
Work perfectly!

File: src/components/Login/index.tsx [hunk 3]
```diff
@@ -50,12 +75,18 @@ function LoginModalContent(props: Props) {
         <ModalCloseButton role="closeButton" />
         <ModalBody>
           <Flex flexDir="column" justifyContent="space-around" gap="25px" paddingX="20px">
-            <Button variant="loginBtn" position="relative" ref={initialRef}>
+            <Button
+              onClick={() => handleLogin()}
```
Comment (Fuchih) [inline src/components/Login/index.tsx:79]: if you don't pass any  props, you can simples use "onClick={handleLogin}" 
you don't need need to change it,  just  a reminder

File: src/components/Login/index.tsx [hunk 4]
```diff
@@ -16,15 +16,47 @@ import MainLogoSvg from "@/assets/svg/CourtCanva-main-LOGO.svg";
...
+  updateLoginData: (data: any) => void;
 }
 
 function LoginModalContent(props: Props) {
   const initialRef = React.useRef(null);
+  const dispatch = useDispatch();
+  // Send request to backend after the request from front-end has been approved by Google
+  /* istanbul ignore next */
+  const handleSuccess = (codeResponse: any) => {
+    api(process.env.NEXT_PUBLIC_API_BASE_URI + "/auth/google"!, {
```
Comment (kritomomo) [inline src/components/Login/index.tsx:38]: You have baseUrl in axiosInstance
image

File: src/components/Login/index.tsx [hunk 5]
```diff
@@ -50,12 +82,18 @@ function LoginModalContent(props: Props) {
         <ModalCloseButton role="closeButton" />
         <ModalBody>
           <Flex flexDir="column" justifyContent="space-around" gap="25px" paddingX="20px">
-            <Button variant="loginBtn" position="relative" ref={initialRef}>
+            <Button
+              onClick={() => handleLogin()}
```
Comment (kritomomo) [inline src/components/Login/index.tsx:86]: image

**Pull request merged**